.github/workflows/snyk-scan.yml

```diff
@@ -90,7 +144,18 @@ jobs:
           SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
         with:
           image: ${{ env.IMAGE_NAME }}
-          args: --severity-threshold=high --file=docker/${{ matrix.sdk }}/Dockerfile.debian
+          args: --severity-threshold=high --file=docker/${{ matrix.sdk }}/Dockerfile.debian --platform=linux/${{ matrix.arch }}
+      - name: Filter Sarif File
+        # GitHub Code Scanning does not allow more than 20 runs per file. We
+        # filter out empty runs to avoid this limit. Note that the file needs to
+        # include at least one run.
+        run: |
+          python -c 'import json; sarif = json.loads(open("snyk.sarif").read()); runs = [run for run in sarif["runs"] if len(run["results"]) > 0]; sarif["runs"] = runs if len(runs) > 0 else [sarif["runs"][0]]; open("out.sarif", "w").write(json.dumps(sarif, indent=2))'
```

Comment:
With the duplication it does feel like a Python script in the repo could be valuable, if that's possible.